### PR TITLE
Enhancement of WASI_SDK_PREFIX in toolchain files

### PR DESCRIPTION
**Pull request by lum1n0us**
No need to pass `WASI_SDK_PREFIX` value from the command line. Use `CMAKE_CURRENT_LIST_DIR` to assemble the value of `WASI_SDK_PREFIX` now.

### CODE REVIEW

**Comment by lum1n0us** [comment]
#346

File: wasi-sdk-pthread.cmake [hunk 1]
```diff
@@ -21,6 +21,11 @@ else()
 	set(WASI_HOST_EXE_SUFFIX "")
 endif()
 
+# When building from source, WASI_SDK_PREFIX represents the generated directory
+if(NOT WASI_SDK_PREFIX)
+  set(WASI_SDK_PREFIX ${CMAKE_CURRENT_LIST_DIR}/../../)
```
Comment (sbc100) [inline wasi-sdk-pthread.cmake:26]: Maybe use a tab for indentation like we already do above?

**Review by sbc100 — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -84,6 +84,20 @@ includes/libraries/etc. The `--sysroot=...` option is not necessary if
 `WASI_SDK_PATH` is `/opt/wasi-sdk`. For troubleshooting, one can replace the
 `--sysroot` path with a manual build of [wasi-libc].
 
+### Integrating with a CMake build system
+
+Use a toolchain file to setup the *wasi-sdk* platform.
+
+```
+$ cmake -DCMAKE_TOOLCHAIN_FILE=${wasi_sdk_home}/share/cmake/wasi-sdk.cmake -S . -B build
```
Comment (sbc100) [inline README.md:92]: How about just `cmake -DCMAKE_TOOLCHAIN_FILE=${wasi_sdk_home}/share/cmake/wasi-sdk.cmake ...`  .. All those other arguments are very project-specific.

Comment (sbc100) [reply README.md:92]: Also, can you use `WASI_SDK_PATH` rather than `wasi_sdk_home` for consistency with the existing example code above?

**Pull request merged**